src/hoc/withRheostat.js

```diff
@@ -105,6 +105,23 @@ const withRheostat = (ChartCompo = null) => {
       );
     }
 
+    componentDidUpdate(prevProps) {
+      if (
+        prevProps.values !== this.props.values &&
+        this.state.handlePos.length === this.props.values.length
+      ) {
+        this.state.handlePos.forEach((pos, index) => {
```

Comment:
Why are we modifying state directly here? Shouldn't this be done with a setState?